src/main/kotlin/com/hibob/academy/resource/PetsResource.kt

```diff
@@ -75,4 +75,27 @@ class PetsResource (private val petService: PetService){
         val petTypeCountMap = petService.getPetTypesAmount(companyId)
         return Response.ok(petTypeCountMap).build()
     }
+
+    @POST
+    @Path("/owner/{ownerId}/company/{companyId}/pets/adopt")
+    fun adoptPets(
+        @PathParam("ownerId") ownerId: Long,
+        @PathParam("companyId") companyId: Long,
+        petIds: List<Long>
+    ): Response {
+        val petsAdoptedCount = petService.adoptMultiplePets(ownerId, petIds, companyId)
+        return Response.ok("$petsAdoptedCount pets were successfully adopted.").build()
+    }
+
+    @POST
+    @Path("/owner/{ownerId}/company/{companyId}/pets/create")
+    fun createMultiplePets(
```

Comment:
Why do you need owner id as path param in multiple pets creation?  They don't have to belong to same owner
In addition isn't owner a part of the pet itself which you can get in the petRecords?